dts/bindings/retained_mem/microcrystal,rv3028-retmem.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 ANITRA system s.r.o.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Micro Crystal RV3028 RTC User RAM
+
+  There are two free RAM bytes for general use.
+
+  Example configuration:
+    rtc: rv3028@52 {
+          compatible = "microcrystal,rv3028";
+          status = "okay";
+          reg = <0x52>;
+          backup-switch-mode = "disabled";
+
+          rtc_retmem: rv3028-retmem {
+                  compatible: "microcrystal,rv3028-retmem";
+                  status = "okay";
+          };
```

Comment:
this should, like the eeprom above, have a reg for offset and size

Reply:
Originally, I didn’t think it would be useful when there were only two bytes available to use. If I were to add the reg property, I’d like some advice on what base address to use specifically. The EEPROM and RAM in the RTC have separate address spaces, and I want to avoid collisions between the unit addresses of the child nodes. User EEPROM occupies addresses 0x00 to 0x2A, while User RAM has addresses 0x1F and 0x20.

Reply:
you use the address space that each occupies, so 0x1f to 0x20, they are separate nodes

Reply:
yes but the child nodes then might have the same unit address in devicetree. however since the unit addresses are optional, this might be a non-issue...

Reply:
Looking at the memory layout, the EEPROM is in a different address space than RAM, I would use the address of the control registers in RAM 0x30 to 0x37 for the EEPROM node (since this is what the driver interacts with):
```
          rtc_eeprom: rv3028-eeprom@30 {
                  compatible: "microcrystal,rv3028-eeprom";
                  status = "okay";
                  reg = <0x30 0x7>;
          };
```

Reply:
> I would use the address of the control registers in RAM 0x30 to 0x37 for the EEPROM node

Hmm... this would prevent configuring the area used in eeprom

Reply:
Oh, you're right :) should the value of the reg property be constant in the binding?

I'll add a sample.

Reply:
@bjarki-andreasen, I followed @nordicjm's suggestion and used the `reg` property to define offset and size for both nodes. It works, dts scripts do not complain, but it still doesn't feel right to me. Neither EEPROM nor RAM addresses belong in the parent node's address space. Wouldn't it be better to use some custom properties that are not mapped to unit addresses?

Here's how the current config looks:
```
&i2c {
  ext_rtc: rv3028@52 {
      compatible = "microcrystal,rv3028";
      reg = <0x52>;
      backup-switch-mode = "disabled";
      status = "okay";

      #address-cells = <1>;
      #size-cells = <1>;

      rtc_eeprom: rv3028-eeprom@0 {
          compatible = "microcrystal,rv3028-eeprom";
          reg = <0x0 0x2b>;
          status = "okay";
      };

      rtc_retmem: rv3028-retmem@1f {
          compatible = "microcrystal,rv3028-retmem";
          reg = <0x1f 0x2>;
          status = "okay";
      };
  };
};
```

Reply:
@decsny I'd appreciate your thoughts on this

Reply:
can you clarify what is your concern?